Merge dev branch into master

pizzalemon opened the pull request:
The Ground Station is in a much different state than it is in the master branch, so it should be merged in. I suggest that after this, we merge dev into master every time we have a fairly good new feature or section of work which distinctly differentiates it from the older version.

pizzalemon committed:
Add Auvsi Suas .proto Python Translations

krishnans2006 committed:
Merge pull request #28 from KrishnanS2006/dev

Updated Flight Map coordinates and Zoom

krishnans2006 committed:
Added Interop Server Connection at Server Initialization

pizzalemon committed:
Fix flight quick data handling and display

pizzalemon committed:
Merge #30 to fix quick data processing

Fix flight quick data handling and display

pizzalemon committed:
Remove unnecessary dropdown scroll bars

pizzalemon committed:
Create custom radio buttons

pizzalemon committed:
Switch radio buttons in flight plan to custom radio buttons

krishnans2006 committed:
Run fixes, Added home route

krishnans2006 committed:
URL Remap /hello -> /

krishnans2006 committed:
Basic Pylint Formatting Applied to all custom Python files in '/' and '/server'.

krishnans2006 committed:
Merge pull request #29 from KrishnanS2006/dev

Added Interop Server Connection at Server Initialization

pizzalemon committed:
Merge pull request #31 for some UI Fixes

Small UI Fixes

krishnans2006 committed:
Added support for ODLC submissions

krishnans2006 committed:
Merge pull request #33 from KrishnanS2006/fix-formatting

Basic Pylint Formatting Applied to all custom Python files

pizzalemon committed:
Add mission data pages to server

pizzalemon committed:
Move Flight Plan into a Flight Data tab

pizzalemon committed:
Add ability to add and display all mission waypoints
- Cannot add obstacles

pizzalemon committed:
Automatically load mission data into flight map

pizzalemon committed:
Remove polygons from map
- change search grid color from pink to polygon color (orange)

pizzalemon committed:
Merge branch 'dev' into client-mission

krishnans2006 committed:
ODLC Save and Load to file

pizzalemon committed:
Add lat-long position to waypoint and obstacle tooltips

pizzalemon committed:
Add ability to change custom radio button color

pizzalemon committed:
Color code radio buttons in Flight Plan

krishnans2006 committed:
ODLC Object Editing before Object Submission

krishnans2006 committed:
ODLC Object Image Access via API Route and Base64 Bytes String

krishnans2006 committed:
Merge remote-tracking branch 'origin/odlc-submission' into dev

krishnans2006 committed:
Started coding incomplete functions in Dummy MAV Handler

pizzalemon committed:
Merge PR for Flight Map Client Mission Loading

Flight Map Client Mission Loading

pizzalemon committed:
Fix adding waypoints stored in an array for flight map

pizzalemon committed:
Remove polygon code artifacts

pizzalemon committed:
Add plane telemetry page to backend

krishnans2006 committed:
Merge pull request #34 from KrishnanS2006/odlc-submission

ODLC submissions

pizzalemon committed:
Update Readme

pizzalemon committed:
Merge pull request for README update

Update Readme

pizzalemon committed:
Update Client Packages

pizzalemon committed:
Add plane position marker to Flight Map
- Also displays the heading of the plane

krishnans2006 committed:
Simplified ODLC integration + Added map submission support

pizzalemon committed:
Merge #41 from pizzalemon/GroundStation for Client Map Improvements

Client Map Improvements

krishnans2006 commented:
I agree with this - however, we should do this after the most recent PRs are merged in (including mine about map submissions, which is about to be created). This is generally a good idea, since `master` is the default branch, and people who might start working with GroundStation might not know to use the `dev` branch instead.

krishnans2006 committed:
Merge branch 'dev' of github.com:tj-uav/GroundStation into dev

krishnans2006 committed:
Merge branch 'dev' into interop-changes

krishnans2006 committed:
Remove space + Add filename input

krishnans2006 committed:
Merge pull request #43 from KrishnanS2006/interop-changes

Added Map Submission

krishnans2006 committed:
Backend Refactor - added error handling, status codes, accurate and consistent responses, method synchronization, and application class

krishnans2006 committed:
Error handling changes

krishnans2006 committed:
Merge branch 'dev' of github.com:tj-uav/GroundStation into dev

krishnans2006 committed:
Re-add sample images

krishnans2006 committed:
Style frontend Quick Data page

krishnans2006 committed:
Merge pull request #54 from KrishnanS2006/application-class

Backend Refactor

pizzalemon commented:
I think we should just close this, especially since the dev branch has become the main branch.

krishnans2006 commented:
I still feel like it is good practice to merge it back into `main` after some major changes are complete - I only set the default branch to `dev` to help with frequent development, and I'll most probably reset it to `main` once we're pretty much done with GroundStation. So it is definitely a good idea, just probably not that relevant right now.

krishnans2006 committed:
Logging (#59)

Added logging to two files + stderr

pizzalemon committed:
ODLC Submission Page (#62)

* ODLC Submission Page

* Fix merge

* Some ODLC Submission Features

* ODLC Submission Page

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

krishnans2006 committed:
Image handler (#61)

* Socket connection to FlightSoftware for handling images

* Better image logging, failsafe for connection loss, code refactor

* Migrate to SocketIO, Fix logging/requirements

* Fix config file/usage

* Fix base64 conversion + logging filenames

* Fix log level + dummy image location

krishnans2006 committed:
Update Flight Data with new interface

pizzalemon committed:
Fixed logging, Separated UAV and UGV Data on frontend

krishnans2006 committed:
Image Handler hotfix - buffer size + image randomization

krishnans2006 committed:
Quick ImageHandler fixes

krishnans2006 committed:
Reformat filesystem for multiple flight handlers, add dummy code for uav sim handler

krishnans2006 committed:
API Routes for (untested) UAV/UGV Handlers

krishnans2006 committed:
Added support for re-initiating Interop Connection on connection loss

pizzalemon committed:
Offline AUVSI SUAS Map

krishnans2006 committed:
UGV stats and map position correctly displayed

krishnans2006 committed:
Modify UAV Prod to accept data from Ardupilot SITL Simulation, Fix bugs/design/organization

dependabot[bot] committed:
Bump url-parse from 1.5.3 to 1.5.10 in /client (#76)

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.5.3 to 1.5.10.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.5.3...1.5.10)

---
updated-dependencies:
- dependency-name: url-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump follow-redirects from 1.14.4 to 1.14.8 in /client (#72)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.14.4 to 1.14.8.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.14.4...v1.14.8)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump nanoid from 3.1.30 to 3.2.0 in /client (#70)

Bumps [nanoid](https://github.com/ai/nanoid) from 3.1.30 to 3.2.0.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.1.30...3.2.0)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump engine.io from 4.1.1 to 4.1.2 in /client (#65)

Bumps [engine.io](https://github.com/socketio/engine.io) from 4.1.1 to 4.1.2.
- [Release notes](https://github.com/socketio/engine.io/releases)
- [Changelog](https://github.com/socketio/engine.io/blob/4.1.2/CHANGELOG.md)
- [Commits](https://github.com/socketio/engine.io/compare/4.1.1...4.1.2)

---
updated-dependencies:
- dependency-name: engine.io
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

krishnans2006 committed:
Organize API into Blueprints, Fix SITL for Prod UAV Handler, Fix bugs

krishnans2006 committed:
Re-organize packages with __init__.py, Use Pylint Google Style Guide

krishnans2006 committed:
Style UAV and UGV, Set up Dummy UAV

Co-authored-by: william blakc <williamblack7586@gmail.com>

krishnans2006 committed:
Implement most Action buttons on frontend

krishnans2006 committed:
Update UGV Icon, Add important stats to Actions page, re-order and update Actions page

krishnans2006 committed:
Fix Dummy UAV to work as a replacement for Prod UAV, fix bugs/typos

krishnans2006 committed:
Fix UAV Mission Actions, Add API Routes and Action Buttons for UGV

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6 in /client (#77)

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

krishnans2006 committed:
Re-organize app blueprints and handlers, remove groundstation function-map, add auto-logging to all necessary functions, implement untested methods for some setup functions

pizzalemon committed:
Fix config file

pizzalemon committed:
Frontend Code Cleanup

krishnans2006 committed:
Implement route calling for all actions, update in-progress routes to not throw errors

krishnans2006 committed:
Update Image retreival to use new HTTP-based system + more

More: Use better type hints for GroundStation-dependent classes, attempt filter for groundstation vs autopilot logs, other bugfixes

pizzalemon committed:
Uavhandler (#4)

* Fix backend config file

* Update react-leaflet to 3.2.5

* Interface fixes; code cleanup

* Client code cleanup and restructure

* Add map getter script

* Add modal UI component

* Backend connection popup

* More merging problems

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

krishnans2006 committed:
Update backend command routes and add route to generate mission file

krishnans2006 committed:
Update backend command routes and add route to generate and export mission file

krishnans2006 committed:
Add route for exporting mission file

pizzalemon committed:
Fix offline map

pizzalemon committed:
Backend Fixes

- Route return fix
- Route file handling
- Image file handling

pizzalemon committed:
Fix ODLC changes reseting on backend query

pizzalemon committed:
User map path planning

pizzalemon committed:
ODLC Long List Loading Optimization

pizzalemon committed:
Fix path discard removing whole queried path, plus insertion and conversion fixes

pizzalemon committed:
Fix path discard removing whole queried path

krishnans2006 committed:
Add telemetry logging, more log viewing routes

krishnans2006 committed:
Hotfixes

More config options for waypoints
Temporarily changed Flight Map Center Pos to FARM RC field
Other bug fixes

pizzalemon committed:
Change Map to Satellite Map

pizzalemon committed:
Fix Polyline Decorator Arrow

pizzalemon committed:
Make Interop Points Non-draggable

pizzalemon committed:
Default Altitude Button, Reset Path Button

pizzalemon committed:
Logs Long List Optimization

pizzalemon committed:
Update Slippy Map Getter Url

pizzalemon committed:
Home and Jump Path Waypoints (In Progress)

krishnans2006 committed:
More hotfixes to backend

krishnans2006 committed:
Last-minute GroundStation fixes before leaving for competition

 - Allow unit switching on click for altitude and speed (to fulfill competition requirements)
 - Add global altitude for UAV (to fulfill competition requirements)
 - Allow Mode Switching for UGV (as an extra option to terminate drive)
 - Remove wait_ready feature for both UAV and UGV (in case of long initialization)
 - Fix Interop Handler to use MSL instead of AGL
 - Add higher zoom functionality by stretching tiles on map
 - Add finer zoom controls on map
 - Format buttons on Actions page
 - Fix Land Command to use Home Location
 - Update mode buttons displayed (more important buttons are now displayed instead)
 - Add option to terminate flight using AFS_TERMINATE
 - Remove unnecessary abort features

krishnans2006 committed:
Reformat with Black

No functionality changes

krishnans2006 committed:
Final changes to GS

 - update slippy map getter, get satellite maps for comp
 - allow (buggy) version of generating path from competition waypoints
 - fix bugs

Co-authored-by: William Black <williamblack7586@gmail.com>

krishnans2006 committed:
Hotfixes:

 - Stop infinite logging on Interop connection loss
 - Add  to dummy UAV handler

pizzalemon committed:
Remove antenna tracker

pizzalemon committed:
Add actions page plane altitude units conversion

pizzalemon committed:
Add tooltips to Quick, Actions, and Submissions

krishnans2006 committed:
Add GitHub workflow and pre-commit hook for black auto-formatting

krishnans2006 committed:
Black refactor of backend

krishnans2006 committed:
Exclude auvsi_suas directory from hooks + more fixes

krishnans2006 committed:
Merge pull request #105 from KrishnanS2006/dev

Add automations for Black refactoring

dependabot[bot] committed:
Bump terser from 4.8.0 to 4.8.1 in /client

Bumps [terser](https://github.com/terser/terser) from 4.8.0 to 4.8.1.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

krishnans2006 committed:
Merge pull request #106 from tj-uav/dependabot/npm_and_yarn/client/terser-4.8.1

Bump terser from 4.8.0 to 4.8.1 in /client

krishnans2006 committed:
Fix typo in "retrieve", add timeout to arming/disarming, add UGV write route for setting destination (untested)

krishnans2006 committed:
Add proper wait-for system on connecting and parameter methods, fix none-setting in initializers of handlers, remove bad files

krishnans2006 committed:
Attempt to copy pre-commit config

krishnans2006 committed:
Add DummyInteropHandler to decouple from interop (temporarily)
Fix attribute definition formatting in InteropHandler
Fix typos

krishnans2006 committed:
Restructure filesystem (utils package)

krishnans2006 committed:
Fix wait_for_param_load, move handlers.utils to utils.decorators

dependabot[bot] committed:
Bump node-forge and react-scripts in /client

Bumps [node-forge](https://github.com/digitalbazaar/forge) and [react-scripts](https://github.com/facebook/create-react-app/tree/HEAD/packages/react-scripts). These dependencies needed to be updated together.

Updates `node-forge` from 0.10.0 to 1.3.1
- [Release notes](https://github.com/digitalbazaar/forge/releases)
- [Changelog](https://github.com/digitalbazaar/forge/blob/main/CHANGELOG.md)
- [Commits](https://github.com/digitalbazaar/forge/compare/0.10.0...v1.3.1)

Updates `react-scripts` from 4.0.3 to 5.0.1
- [Release notes](https://github.com/facebook/create-react-app/releases)
- [Changelog](https://github.com/facebook/create-react-app/blob/main/CHANGELOG-4.x.md)
- [Commits](https://github.com/facebook/create-react-app/commits/react-scripts@5.0.1/packages/react-scripts)

---
updated-dependencies:
- dependency-name: node-forge
  dependency-type: indirect
- dependency-name: react-scripts
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

krishnans2006 committed:
Add logs, fix protobuf version for compatibility with dronekit

JunePikner committed:
Logs page scrolling fix

Co-authored-by: Joshua <joshuapikner@gmail.com>

krishnans2006 committed:
Telem stuff

dependabot[bot] committed:
Bump loader-utils from 2.0.1 to 2.0.3 in /client

Bumps [loader-utils](https://github.com/webpack/loader-utils) from 2.0.1 to 2.0.3.
- [Release notes](https://github.com/webpack/loader-utils/releases)
- [Changelog](https://github.com/webpack/loader-utils/blob/v2.0.3/CHANGELOG.md)
- [Commits](https://github.com/webpack/loader-utils/compare/v2.0.1...v2.0.3)

---
updated-dependencies:
- dependency-name: loader-utils
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump socket.io-parser from 4.0.4 to 4.0.5 in /client

Bumps [socket.io-parser](https://github.com/socketio/socket.io-parser) from 4.0.4 to 4.0.5.
- [Release notes](https://github.com/socketio/socket.io-parser/releases)
- [Changelog](https://github.com/socketio/socket.io-parser/blob/main/CHANGELOG.md)
- [Commits](https://github.com/socketio/socket.io-parser/compare/4.0.4...4.0.5)

---
updated-dependencies:
- dependency-name: socket.io-parser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump minimatch and recursive-readdir in /client

Bumps [minimatch](https://github.com/isaacs/minimatch) and [recursive-readdir](https://github.com/jergason/recursive-readdir). These dependencies needed to be updated together.

Updates `minimatch` from 3.0.4 to 3.1.2
- [Release notes](https://github.com/isaacs/minimatch/releases)
- [Commits](https://github.com/isaacs/minimatch/compare/v3.0.4...v3.1.2)

Updates `recursive-readdir` from 2.2.2 to 2.2.3
- [Release notes](https://github.com/jergason/recursive-readdir/releases)
- [Changelog](https://github.com/jergason/recursive-readdir/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jergason/recursive-readdir/commits/v2.2.3)

---
updated-dependencies:
- dependency-name: minimatch
  dependency-type: indirect
- dependency-name: recursive-readdir
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

krishnans2006 committed:
Document server (#112)

* Rename app to apps

* Typehint files in / and /utils

* Add mypy (still finicky)

* Fix mypy pre-commit hook, remove unused/obsolete files, minor type hint fixes/ignore import fixes

* Fix mypy issues

* Add pre-commit.ci configuration, remove black workflow

krishnans2006 committed:
Bulk removal of UGV and Interop, combine Quick and Actions tab into Main

krishnans2006 committed:
Servo tab with read-only min/max/trim values

krishnans2006 committed:
Revert map center position to FARM RC

krishnans2006 committed:
Merge pull request #111 from JoshuaPikner/dev

fixed up the logs page; no longer cuts off the scrolling and now refr…

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 22.6.0 → 22.10.0](https://github.com/psf/black/compare/22.6.0...22.10.0)

krishnans2006 committed:
Merge pull request #120 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

krishnans2006 committed:
Folder for archive, remove log files

krishnans2006 committed:
Run `npm audit fix --force` to fix issues with SSL/package versions

Slightly modified package.json, regenerated package-lock.json

krishnans2006 committed:
Merge pull request #115 from KrishnanS2006/telem-stuff

Fix Telemetry logging and parsing

krishnans2006 committed:
Giant filesystem revamp (move all static files into assets, remove unnecessary folders), add distance from Home, fix filepath references for cross-OS compatibility, style Main data page, fix sample.config.json for new config structure

krishnans2006 committed:
Minor pre-commit and requirements changes

krishnans2006 committed:
Revamp image API to support features from tj-uav/FlightSoftware#12

pizzalemon committed:
Mission Planning

dependabot[bot] committed:
Bump decode-uri-component from 0.2.0 to 0.2.2 in /client

Bumps [decode-uri-component](https://github.com/SamVerschueren/decode-uri-component) from 0.2.0 to 0.2.2.
- [Release notes](https://github.com/SamVerschueren/decode-uri-component/releases)
- [Commits](https://github.com/SamVerschueren/decode-uri-component/compare/v0.2.0...v0.2.2)

---
updated-dependencies:
- dependency-name: decode-uri-component
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

krishnans2006 committed:
Update README.md for new deployment instructions

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 22.10.0 → 22.12.0](https://github.com/psf/black/compare/22.10.0...22.12.0)

dependabot[bot] committed:
Bump json5 from 1.0.1 to 1.0.2 in /client

Bumps [json5](https://github.com/json5/json5) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/json5/json5/releases)
- [Changelog](https://github.com/json5/json5/blob/main/CHANGELOG.md)
- [Commits](https://github.com/json5/json5/compare/v1.0.1...v1.0.2)

---
updated-dependencies:
- dependency-name: json5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

krishnans2006 committed:
DummyUAVHandler + other bugfixes and typehints

Other bugfixes: Add many previously ignored files/folders for better initial setup, add pre-commit to requirements

krishnans2006 committed:
Frontend hotfixes

Downgrade react-scripts to 4.0.3 to fix "Invalid options object" error, add "--openssl-legacy-provider" to fix "digital envelope routines::unsupported" error

krishnans2006 committed:
Hotfixes: flight plan changing midflight bug and port bug

pizzalemon committed:
Fix 2023-update and mission-planning Merge Errors

krishnans2006 committed:
Log file button hotfix

Also package-lock.json for some reason

krishnans2006 committed:
Merge pull request #125 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 22.12.0 → 23.3.0](https://github.com/psf/black/compare/22.12.0...23.3.0)

krishnans2006 committed:
Merge pull request #129 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

krishnans2006 committed:
Add battery monitors with message listeners, update parameters file

krishnans2006 committed:
Add servo outputs with message listeners, fix styling of Servo tab

krishnans2006 committed:
Add error handling in httpget, catch 409's when accessing parameter data in servo tab

tjuav committed:
React App UI changes (favicons, titles, descriptions)

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Fix root requirements.txt file

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Add shell scripts (and other files) to automate running the simulation and GS

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Add CODEOWNERS file

krishnans2006 committed:
Merge pull request #124 from tj-uav/dependabot/npm_and_yarn/client/decode-uri-component-0.2.2

Bump decode-uri-component from 0.2.0 to 0.2.2 in /client

krishnans2006 committed:
Merge pull request #126 from tj-uav/dependabot/npm_and_yarn/client/json5-1.0.2

Bump json5 from 1.0.1 to 1.0.2 in /client

tjuav committed:
Hotfix: Second battery output on Dummy UAV Handler

Allow for passing a blank port in run-gs.sh

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Don't open the browser when starting with run-gs.sh

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Add `util` folder to handle setup and config scripts/files. Write documentation on setting up SITL and GroundStation to run as desktop apps.

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Make UAV black-and-white when disarmed and colored when armed

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Change default altitude to 125ft

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Remove artifacts from last year in FlightMap.js

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Remove artifacts from last year in FlightPlanToolbar.js

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Move all getters into props.getters and all setters into props.setters

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Streamline map icons and colors, add flight and airdrop boundaries from rules

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Add home location data (but not to map)

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

tjuav committed:
Fix new waypoints not being unsaved

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>
Co-authored-by: William Black <williamblack7586@gmail.com>

tjuav committed:
Fix altitude input not allowing input

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>
Co-authored-by: William Black <williamblack7586@gmail.com>

krishnans2006 committed:
Get home icon to work, also add home location functionality to DummyUAV

JunePikner committed:
Adds a new dynamic icon that shows estimated water bottle drop position, currently just a dummy pos of 20, 20, but can be easily changed later.

JunePikner committed:
Adds a new dynamic icon that shows estimated water bottle drop position, currently just a dummy pos of 20, 20, but can be easily changed later.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

krishnans2006 committed:
Add latitude and longitude controls to waypoints, fix disabled save button when editing popup values

krishnans2006 committed:
Path popup streamlining (add title and datatype for future use), fix pathSaved on click

krishnans2006 committed:
Default jump waypoint to 999 repeats

krishnans2006 committed:
Add custom popup for jump waypoints (in progress)

krishnans2006 committed:
Finish custom popup for jump waypoints

JunePikner committed:
Just adding in some placeholder equations, they aren't very good at all but it's better than a static dummy

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

tjuav committed:
Home location hotfix

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Update caniuse to get rid of annoying warning message

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

tjuav committed:
Default initial placement mode to disabled

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

krishnans2006 committed:
Add Radius (p3) for loiters, fix popup title naming, fix waypoint number in Main

krishnans2006 committed:
Add # of Turns (p1) and Time (p1) for turn and time loiters

krishnans2006 committed:
Don't update unnecessarily

krishnans2006 committed:
UI revamp!!! (On Main and Map tabs)

krishnans2006 committed:
Wait for vehicle attributes to be populated before fetching data

krishnans2006 committed:
Fix Leaflet Layer Control names for home location and bottle drop

krishnans2006 committed:
UI Revamp part 2 to actually show MSL/knots data

krishnans2006 committed:
Fix glitchy default altitude editing

krishnans2006 committed:
Use "knots" instead of "kn"

krishnans2006 committed:
Hide distracting waypoints when not in AUTO

krishnans2006 committed:
Modal to avoid misclicking Terminate button

krishnans2006 committed:
More warnings on terminate modal

krishnans2006 committed:
Support pypoetry with poetry.lock and pyproject.toml, reformat requirements.txt

krishnans2006 committed:
Update slippy_map_getter.py useful locations

krishnans2006 committed:
Fix offline map issues with a bunch of try-catch

krishnans2006 committed:
Even more large-scale tiles

krishnans2006 committed:
Merge pull request #119 from KrishnanS2006/2023-update

Update GroundStation for new AUVSI SUAS Rules

krishnans2006 committed:
Remove dronekit-sitl from requirements.txt

krishnans2006 committed:
Add rolling logs to server logging setup

krishnans2006 committed:
Fix rolling log setup on server

krishnans2006 committed:
Basic rolling log display on Main, comment out Logs tab

tjuav committed:
Updated "." to represent "/dev/ttyUSB0"

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>
Co-authored-by: Joshua <joshuapikner@gmail.com>

krishnans2006 committed:
Actual Log styling

krishnans2006 committed:
Log regex replacements for dates and categories

krishnans2006 committed:
Update home location to competition requirement, add "Set Home" button to Map tab

krishnans2006 committed:
Update AUVSI_SUAS simulation altitude

krishnans2006 committed:
Update map center and zoom for competition

krishnans2006 committed:
Update home location altitude to ground height

krishnans2006 committed:
Update avalon.param with updated params from plane

krishnans2006 committed:
Add new plane.json and plane.param to keep consistent naming scheme

krishnans2006 committed:
Add params_to_json.py to convert raw parameter files to json files for server processing

krishnans2006 committed:
Update util scripts to handle incorrect working directory

krishnans2006 committed:
Rename `util` to `scripts`

krishnans2006 committed:
Export Ardupilot's complete parameters list for offline access

krishnans2006 committed:
Make Boundary icons blank (with new NoIcon())

krishnans2006 committed:
Adjust boundary line weights

krishnans2006 committed:
Add distance calculation mode to Map tab, refactor firstJump and firstPoint states

krishnans2006 committed:
Rename mode to placementType, resize dropdowns

krishnans2006 committed:
Rename jumpClick to doubleClick

krishnans2006 committed:
Fix new waypoint numbering, refactor doubleClick

krishnans2006 committed:
Refactor waypointClick (formerly doubleClick)

krishnans2006 committed:
Add help text for jumps

krishnans2006 committed:
Default altitude to 250

krishnans2006 committed:
Update help text for all placement modes

krishnans2006 committed:
Remove console.log artifact

krishnans2006 committed:
Adjust Tooltip display and decimal places

krishnans2006 committed:
Adjust marker display names to differentiate LayersControl and Tooltip

krishnans2006 committed:
Universal 8-digit lat/lng in Tooltips

krishnans2006 committed:
Fix `Cannot read properties of undefined (reading 'toFixed'` when lat or lng is not defined

krishnans2006 committed:
Proper dropdown initial values in Map

krishnans2006 committed:
Loading logs placeholder

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.3.0 → 23.7.0](https://github.com/psf/black/compare/23.3.0...23.7.0)

krishnans2006 committed:
Merge pull request #140 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

tjuav committed:
Fix run-gs script to run server on 0.0.0.0

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.7.0 → 23.9.1](https://github.com/psf/black/compare/23.7.0...23.9.1)

krishnans2006 committed:
Merge pull request #142 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

tjuav committed:
Slippy map getter reformat

Co-authored-by: Krishnan Shankar <krishnans2006@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 23.9.1 → 23.10.0](https://github.com/psf/black/compare/23.9.1...23.10.0)

krishnans2006 committed:
Merge pull request #143 from tj-uav/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

krishnans2006 committed:
Hotfix: Add transparent icon used in efba794
